tests/core/test_utils.py

```diff
@@ -246,3 +247,26 @@ def test_ignores_if_subobject_is_not_dict(self):
         obtained = dict_subset(input, ["a", "b.c.d", "b.d"])
         expected = dict(a=1, b=dict(c=2, d=3))
         self.assertEqual(obtained, expected)
+
+
+class ParseResourceTest(unittest.TestCase):
+
+    def test_parse_resource(self):
+        bad_inputs = ['none', '/', '/one', 'one/', '/tw/o']
+        good_inputs = ['bid/cid', '/buckets/bid/collections/cid']
+        expected = {
+            'bucket': 'bid',
+            'collection': 'cid'
+        }
+        for input in bad_inputs:
+            error_msg = ("Resources should be defined as "
+                         "'/buckets/<bid>/collections/<cid>' or "
+                         "'<bid>/<cid>'. Got %r" % input)
+
+            with pytest.raises(ValueError) as excinfo:
+                parse_resource(input)
+            self.assertEquals(str(excinfo.value), error_msg)
+
+        for input in good_inputs:
+            parts = parse_resource(input)
+            self.assertEqual(expected, parts)
```

Comment:
Thank you for the review @leplatrem 
I changed the files according to your requests